src/core/StelProjector.cpp

```diff
@@ -152,19 +152,14 @@ void StelProjector::init(const StelProjectorParams& params)
 	zNear = params.zNear;
 	oneOverZNearMinusZFar = 1./(zNear-params.zFar);
 	viewportCenterOffset = params.viewportCenterOffset;
-	viewportXywh = params.viewportXywh;
-	viewportXywh[0] *= devicePixelsPerPixel;
-	viewportXywh[1] *= devicePixelsPerPixel;
-	viewportXywh[2] *= devicePixelsPerPixel;
-	viewportXywh[3] *= devicePixelsPerPixel;
-	viewportCenter = params.viewportCenter;
-	viewportCenter *= devicePixelsPerPixel;
+	viewportXywh = params.viewportXywh * static_cast<int>(devicePixelsPerPixel);
```

Comment:
Scaling factor must not be rounded, rounding will break scales like 150% or 125%, which are often the default or recommended OS settings.